Add AWS Devicefarm job back into CI

sergeichestakov opened the pull request:
# Why

We want test-suite running all the tests on a device in devicefarm after every commit. This PR adds that functionality back into CI.

# How

I added the CI job to run test-suite on AWS back into the `.circleci/config.yml` within the client workflow. I also switched to publishing the tests on a channel based off of the commit SHA rather than the branch name so that we can have separate published versions on the same branch for each new commit. I also updated some of the tests that have been failing, starting with `Basic.js`.

# Test Plan

This should run as a CI job in the new `.circleci/config.yml`. It will fail during the client workflow initially as 27 of the tests do not currently pass. This PR hopes to address that and get it back to green by either fixing the tests or commenting them/assigning them to others for now until we can get them fixed.

Commit:
Publish to commit SHA instead of branch name

Commit:
Add android devicefarm job back into client workflow in CI

Commit:
Combine Basic tests and remove the ones that fail

Commit:
Fix indentation

Commit:
Fix Import test by not using default import

esamelson reviewed (commented):
this looks good so far but let's hold off on merging until we can get it green (or have a clear plan/path to do so).

Commit:
Show the number of tests failed in TestScreen

Commit:
Fix typo in FileSystem that caused test to fail

Commit:
Uncomment and fix remaining FileSystem tests

Commit:
Fix Camera roll permissions in FileSystem test

Commit:
Add Audio recording permissions test

Commit:
Fix Recording tests

staysActiveInBackground is a required attribute on the front end and there was a typo in the isMissingAudioRecordingPermissions function which prevented the tests from running because the module rejected the promise on the basis that the required permissions were not met when they actually were

Commit:
Accept Camera Roll permissions in FileSystem test

Commit:
Move acceptPermissions to beforeAll in FileSystem

sjchmiela commented on packages/expo-av/android/src/main/java/expo/modules/av/AVManager.java:
```diff
@@ -500,7 +500,7 @@ public void runWithVideoView(final VideoView videoView) {
 
   private boolean isMissingAudioRecordingPermissions() {
-    return mModuleRegistry.getModule(Permissions.class).getPermission(Manifest.permission.RECORD_AUDIO) == PackageManager.PERMISSION_GRANTED;
+    return mModuleRegistry.getModule(Permissions.class).getPermission(Manifest.permission.RECORD_AUDIO) != PackageManager.PERMISSION_GRANTED;
```
😅 😅 😅

Commit:
Remove/document the tests that fail in CI

Commit:
Stop running Audio tests in CI

sergeichestakov commented:
Ok after fixing and removing the tests that were failing, all of the tests that run in CI now pass so the client workflow is all green 👍

esamelson reviewed (approved):
this looks good to me! you can go ahead and merge 🚀
